vello_encoding/src/path.rs

```diff
@@ -833,6 +902,174 @@ impl skrifa::outline::OutlinePen for PathEncoder<'_> {
+        self.in_subpath = true;
+
+        self.encoder.move_to(x, y);
+    }
+
+    pub fn line_to(&mut self, x: f32, y: f32) {
+        if self.in_subpath {
+            self.current_subpath.push(Point::new(x as f64, y as f64));
+        }
+
+        self.encoder.line_to(x, y);
+    }
+
+    pub fn quad_to(&mut self, x1: f32, y1: f32, x2: f32, y2: f32) {
+        if self.in_subpath {
+            self.current_subpath.push(Point::new(x2 as f64, y2 as f64));
+        }
+
+        self.encoder.quad_to(x1, y1, x2, y2);
+    }
+
+    pub fn cubic_to(&mut self, x1: f32, y1: f32, x2: f32, y2: f32, x3: f32, y3: f32) {
+        if self.in_subpath {
+            self.current_subpath.push(Point::new(x3 as f64, y3 as f64));
+        }
+
+        self.encoder.cubic_to(x1, y1, x2, y2, x3, y3);
+    }
+
+    pub fn close(&mut self) {
+        if self.in_subpath && !self.current_subpath.is_empty() {
+            self.all_glyph_points
+                .extend_from_slice(&self.current_subpath);
+        }
+
+        self.in_subpath = false;
+        self.current_subpath.clear();
+
+        self.encoder.close(true);
+    }
+
+    pub fn shape(&mut self, shape: &impl Shape) {
+        self.path_elements(shape.path_elements(0.1));
+    }
+
+    pub fn path_elements(&mut self, path: impl Iterator<Item = peniko::kurbo::PathEl>) {
+        use peniko::kurbo::PathEl;
+        for el in path {
+            match el {
+                PathEl::MoveTo(p0) => self.move_to(p0.x as f32, p0.y as f32),
+                PathEl::LineTo(p0) => self.line_to(p0.x as f32, p0.y as f32),
+                PathEl::QuadTo(p0, p1) => {
+                    self.quad_to(p0.x as f32, p0.y as f32, p1.x as f32, p1.y as f32);
+                }
+                PathEl::CurveTo(p0, p1, p2) => self.cubic_to(
+                    p0.x as f32,
+                    p0.y as f32,
+                    p1.x as f32,
+                    p1.y as f32,
+                    p2.x as f32,
+                    p2.y as f32,
+                ),
+                PathEl::ClosePath => self.close(),
+            }
+        }
+    }
+
+    pub fn finish(mut self, insert_path_marker: bool) -> u32 {
+        if self.in_subpath && !self.current_subpath.is_empty() {
+            self.all_glyph_points
+                .extend_from_slice(&self.current_subpath);
+        }
+
+        // compute a single winding for the entire glyph/path
+        if !self.all_glyph_points.is_empty() && !self.winding_stored {
+            let winding = compute_winding(&self.all_glyph_points);
+            let winding_sign = if winding == 0 { -1.0 } else { 1.0 };
+
+            self.path_windings.push(winding_sign);
+            if let Some(tag) = self.encoder.tags.last_mut() {
+                tag.set_winding_incr();
+            }
+
+            self.winding_stored = true;
+        }
+
+        self.encoder.finish(insert_path_marker)
+    }
+}
+
+fn compute_winding(points: &[Point]) -> u8 {
```

Comment:
Unless I'm missing something, this has the possibility of mis-computing area, as I believe it includes a component from the last point of a subpath to the `moveto` opening the next subpath, and this should not be included. I think this logic would be fairly straightforward to accumulate on the fly.